File: src/Web/src/services/interfaces/IServiceLocationService.ts
import type { ServiceResult } from "../../models/baseServiceModels";
import type { ServiceLocationView, ServiceLocationRequest, ServiceLocationReadyRequest } from "../../models/ServiceLocationModels";

export interface IServiceLocationService {
    getListAsync(): Promise<ServiceResult<ServiceLocationView[]>>;
    getByIdAsync(id: string): Promise<ServiceResult<ServiceLocationView>>;
    addAsync(request: ServiceLocationRequest): Promise<ServiceResult<string>>;
    updateAsync(id: string, request: ServiceLocationRequest): Promise<ServiceResult<ServiceLocationView>>;
    deleteAsync(id: string): Promise<ServiceResult<void>>;
    notifyAvailableAsync(id: string, request: ServiceLocationReadyRequest): Promise<ServiceResult<string>>;
}

File: src/Services/Totem.Application/Services/ServiceLocationServices/IServiceLocationService.cs
using Totem.Common.Services;
using Totem.Domain.Models.ServiceLocationModels;
using Totem.SharedKernel.Models;

namespace Totem.Application.Services.ServiceLocationServices
{
	public interface IServiceLocationService
	{
		Task<(Result result, List<ServiceLocationSummary> data)> GetListAsync();
		Task<(Result result, ServiceLocationView data)> GetByIdAsync(Guid Id);
		Task<(Result result, Guid data)> AddAsync(ServiceLocationRequest request);
		Task<Result> UpdateAsync(Guid Id, ServiceLocationRequest request);
		Task<Result> DeleteAsync(Guid Id);

		Task<(Result result, IPasswordView data)> ServiceLocationReadyAsync(Guid serviceLocationId, ServiceLocationReadyRequest request);

		/// <summary>Notifies all clients in the service location group to re-display the current password on the panel.</summary>
		Task<Result> RecallCurrentPasswordAsync(Guid serviceLocationId);

	}
}
